### ISSUE TITLE
Podczas zmiany runy w mieczu runicznym w siedzibie nekromanty wywala z gry

### ISSUE THREAD

**Issue by Adanos** [orcwarrior/Czas_Zaplaty#331]
Podczas zmiany runy w mieczu runicznym w siedzibie nekromanty wywala z gry.

-4-	00:58 Info:  5 U:    Skript: PerFrame: Pre-Init :( .... <zError.cpp,#462>
-4-	00:58 Info:  5 U:    Skript: PerFrame: FrameCallBack_HandleKeystorkes .... <zError.cpp,#462>
-4-	00:58 Info:  5 U:    Skript: FrameCallBack_HandleKeystorkes RuneSwd .... <zError.cpp,#462>
-4-	00:58 Info:  2 U:    PAR: Adressing an empty Instance : C_ITEM.NAME .... <zParser_Symbol.cpp,#365>

**Comment by Adanos** [comment]
Przy każdej zmianie zena.

**Comment by Splash97** [comment]
Wszystkie npc obsługujące systemy są w siedzibie nekromanty? Może brakuje jakiegoś WP. Pamiętam, że w modzie są oginiki, które miały nazwy czarów zamiast imienia. Może to ich brakuje?

**Comment by Adanos** [comment]
Są ogniki. Teleportowałem się na jakiś wp INVISIBLE i byli tam NPC kula ognia itp. xD
Co do zmiany zena, to chodziło mi o każdą zmianę zena, niekoniecznie do nekromanty.
Wydaje mi się, że przy zmianie zena nie przechodzi informacja o włożonej runie.

**Comment by Splash97** [comment]
Runa nie jest zapisywana w zmiennej globalnej?

**Comment by Adanos** [comment]
Szczerze mówiąc, to nie wiem gdzie. xD
Ogólnie to jest jakaś zmienna globalna item, i wywala wtedy, gdy wciskasz klawisz odpowiadający za włożenie runy do miecza. Zmienna item powinna wskazywać na runę, ale na nic nie wskazuje, a na pewno nie na żaden przedmiot. Dokładnie to w tej funkcji KeyEvent_RuneSwd: https://github.com/orcwarrior/Czas_Zaplaty/blob/master/Content/AI/AI_Intern/Controls_Events.d#L13-L42

**Comment by Splash97** [comment]
MEM_CallByString (concatstrings("RUNESWD_EQUIP_RUNECHOICE_",rune_instance));
Czyli ta funkcja wywołuje RUNESWD_EQUIP_RUNECHOICE_InstancjaRuny.
Np. RuneSwd_Equip_RuneChoice_Fireball
Jeśli dobrze rozumiem, npc obsługujący ten system powinien zwrócić włożoną wcześniej runę, która trafiła do jego ekwipunku. Ale w innych światach oni nie otrzymują tych run, więc nie mają jej w ekwipunku. To jest dziwne zrobione, bo dużo prościej byłoby przypisać runę do zmiennej globalnej, niż dawać ją jakiemuś npc, bo w innym świecie npc nie ma tej runy, dlatego chyba w zSpy pojawia się Adressing an empty Instance : C_ITEM.NAME, a mod się crashuje.

**Comment by Adanos** [comment]
No, pewnie tak, wiele jest takich dziwnych rzeczy. Mogę spróbować dać zmienną globalną, zobaczymy, czy się coś zmieni.

**Comment by Adanos** [comment]
Nie wiem, czy zmienna globalna pomoże, skoro problemem jest to, że przy zaznaczeniu nowej runy, którą chce się włożyć, wywala z gry. Problem się też pojawia, gdy zmienimy zena, ale tu chyba pomocne może być zapisanie, wczytanie gry.
Kolejny problem możemy używać miecza runicznego, mimo że nie nauczyliśmy się nim posługiwać. Tak więc uczenie się posługiwania tym mieczem jest bezsensu.

**Comment by Adanos** [comment]
Sprawdziłem. Zapisanie i wczytanie po zmianie zena eliminuje wywalanie z gry. Przynajmniej będzie to jakieś rozwiązanie dla graczy...
